[DOCS-8771] Add Cloud SIEM Security Operational Metrics (#24883)
DataDog/documentation: content/en/security/cloud_siem/security_operational_metrics.md

---
title: Security Operational Metrics
disable_toc: false
further_reading:
- link: "security/cloud_siem/investigate_security_signals"
  tag: "Documentation"
  text: "Investigate Cloud SIEM Security Signals"
- link: "getting_started/dashboards"
  tag: "Documentation"
  text: "Getting started with dashboards"
- link: "getting_started/monitors"
  tag: "Documentation"
  text: "Getting started with monitors"
- link: "metrics/summary/"
  tag: "Documentation"
  text: "Learn more about the Metrics Summary"
---

## Overview

> Cloud SIEM provides security operational metrics to help you determine the effectiveness of your team in responding to and resolving security threats to your cloud environments. These metrics are shown in the out-of-the-box [Cloud SIEM dashboard][1] and also sent in the Cloud SIEM [weekly digest reports][2]. You can also create dashboards and monitors for them.

Reviewer: ```suggestion
Cloud SIEM provides security operational metrics to help you determine the effectiveness of your team in responding to and resolving security threats to your cloud environments. These metrics are shown in the out-of-the-box [Cloud SIEM dashboard][1] and are sent in the Cloud SIEM [weekly digest reports][2]. You can also create dashboards and monitors for them.
```

{{< img src="security/security_monitoring/security_operational_metrics.png" alt="The security operational metrics section of the Cloud SIEM Overview dashboard" style="width:100%;" >}}

## Operational Metrics

`datadog.security.siem_signal.time_to_detect`
: **Name**: Mean Time to Detect (MTTD)
: **Description**:The average time from when a matching log is triggered to when a signal is generated.

`datadog.security.siem_signal.time_to_acknowledge`
: **Name**:  Mean Time to Acknowledge (MTTA)
: **Description**: The average time from when a signal is triggered to when an investigation on the signal begins.

`datadog.security.siem_signal.time_to_resolve`

> : **Name**: Mean Time to Resolve (MTTR)

Reviewer 2: Can we replace all of the "Mean Time" to just "Time". We'd like to refer to the metrics as time to detect, time to acknowledge, etc. 

Mean can be derived from the metric itself.

Author: Removed `Mean`.

> : **Description**:The average time from when a matching log is triggered to when a signal is generated.
> 
> `datadog.security.siem_signal.time_to_acknowledge`
> : **Name**: Time to Acknowledge (TTA)
> : **Description**: The average time from when a signal is triggered to when an investigation on the signal begins.
> 
> `datadog.security.siem_signal.time_to_resolve`
> : **Name**: Time to Resolve (TTR)
> : **Description**: The average time it takes to close a signal from the time when you are first notified of the detection.

Reviewer: ```suggestion
: **Description**: The average time between when a matching log is triggered and when a signal is generated.

`datadog.security.siem_signal.time_to_acknowledge`
: **Name**: Time to Acknowledge (TTA)
: **Description**: The average time between when a signal is triggered and when an investigation on the signal begins.

`datadog.security.siem_signal.time_to_resolve`
: **Name**: Time to Resolve (TTR)
: **Description**: The average time it takes to close a signal starting from the time when you are first notified of the detection.
```

## How the metrics are calculated

The TTD, TTA, and TTR metrics are calculated based on these timestamps:

> 1. The timestamp (`T0`) of the log that triggered a security signal.
> 1. The timestamp (`T1`) of when the signal was generated.
> 1. The timestamp (`T3`) of when the signal status was changed to `under_review`.
> 1. The timestamp (`T4`) of when the signal status was changed to `archived`.

Reviewer: ```suggestion
1. The timestamp (`T0`) of the log that triggers a security signal.
1. The timestamp (`T1`) of when the signal is generated.
1. The timestamp (`T3`) of when the signal status is changed to `under_review`.
1. The timestamp (`T4`) of when the signal status is changed to `archived`.
```

Feel free to ignore, because the original isn't _wrong_, but for some reason the past tense was sounding a little off to me.

Author: @neko-dd Thanks for bringing this up! I bounced between the two versions a couple of times when writing this section. I'm going to go with the present tense.

| Metric                                                                                | How the metric is calculated |
| ------------------------------------------------------------------------------------- | ---------------------------- |

> | Mean Time to Detect (MTTD)<br>`datadog.security.siem_signal.time_to_detect`           | `T1 - T0`                    |

Reviewer 2: Same comment as above. Let's remove "Mean"

Author: Done.

| Time to Acknowledge (TTA)<br>`datadog.security.siem_signal.time_to_acknowledge` | `T2 - T1`                    |
| Time to Resolve (TTR)<br>`datadog.security.siem_signal.time_to_resolve`         | `T3 - T1`                    |

## Explore, visualize, and monitor the metrics

> Use the [Metrics Summary][3] to view details about the operational metrics, such as the tags and assets (dashboards, notebooks, monitors, SLOs) that are using those metrics.

Reviewer: I'm not sure what this sentence means. Can you clarify?

Author: @neko-dd Does this work better?

> Use the [Metrics Summary][3] to see the operational metrics' metadata and tags. You can also see which dashboards, notebooks, monitors, and SLOs are using those metrics.

> Use tags to filter the metrics to specific teams, sources, and environments and then create [dashboards][4] for those metrics to visualize the data. You can also create [monitors][5] to alert you if the metrics exceed a specified threshold.

Reviewer: ```suggestion
Use tags to filter the metrics to specific teams, sources, and environments. You can then create [dashboards][4] for those metrics to visualize the data or create [monitors][5] to alert you if the metrics exceed a specified threshold.
```

: **Description**: The time it takes to close a signal starting from the time when you are first notified of the detection.
: **Metric type**: [DISTRIBUTION][3]

## How the metrics are calculated

The TTD, TTA, and TTR metrics are calculated based on these timestamps:

1. The timestamp (`T0`) of the log that triggers a security signal.
1. The timestamp (`T1`) of when the signal is generated.
1. The timestamp (`T3`) of when the signal status is changed to `under_review`.
1. The timestamp (`T4`) of when the signal status is changed to `archived`.

| Metric                                                                                | How the metric is calculated |
| ------------------------------------------------------------------------------------- | ---------------------------- |
| Time to Detect (TTD)<br>`datadog.security.siem_signal.time_to_detect`           | `T1 - T0`                    |
| Time to Acknowledge (TTA)<br>`datadog.security.siem_signal.time_to_acknowledge` | `T2 - T1`                    |
| Time to Resolve (TTR)<br>`datadog.security.siem_signal.time_to_resolve`         | `T3 - T1`                    |

## Explore, visualize, and monitor the metrics

> Use the [Metrics Summary][4] to view details about the operational metrics, such as the tags and assets (dashboards, notebooks, monitors, SLOs) that are using those metrics.

Author: ```suggestion
Use the [Metrics Summary][3] to see metadata and tags for the operational metrics. You can also see which dashboards, notebooks, monitors, and SLOs are using those metrics.
```
